baseplate/frameworks/pyramid/__init__.py

```diff
@@ -333,7 +333,7 @@ def _on_new_request(self, event: pyramid.events.ContextFound) -> None:
         if request.matched_route:
             endpoint = (
                 request.matched_route.pattern
-                if request.matched_route.pattern
+                if hasattr(request.matched_route, "pattern")
```

Comment:
Should this be?
```
if hasattr(request.matched_route, "pattern") and request.matched_route.pattern:
```

With the old behavior, we would use `request.matched_route.name` if `request.matched_route.pattern` were `None`. Now, we will use `request.matched_route.pattern` any time it's present (even if it's `None`).

Reply:
Good call, thanks for the extra thought!